src/coreclr/src/vm/gcenv.ee.cpp

```diff
@@ -763,14 +763,15 @@ void GCToEEInterface::DiagWalkSurvivors(void* gcContext, bool fCompacting)
 #endif //GC_PROFILING || FEATURE_EVENT_TRACE
 }
 
-void GCToEEInterface::DiagWalkLOHSurvivors(void* gcContext)
+void GCToEEInterface::DiagWalkUOHSurvivors(void* gcContext, int gen)
 {
 #if defined(GC_PROFILING) || defined(FEATURE_EVENT_TRACE)
     if (ShouldTrackSurvivorsForProfilerOrEtw())
     {
         size_t context = 0;
         ETW::GCLog::BeginMovedReferences(&context);
-        GCHeapUtilities::GetGCHeap()->DiagWalkSurvivorsWithType(gcContext, &WalkMovedReferences, (void*)context, walk_for_loh);
+        walk_surv_type type = (gen == 3) ? walk_for_loh : walk_for_poh;
```

Comment:
> (gen == 3) [](start = 29, length = 11)

it seems very odd to have a hard coded value here. 

you could just have DiagWalkSurvivorsWithType take an additional parameter which is the generation.

Reply:
I probably wanted to minimize changes. Using literal indeed looks strange now. Will change.